Sources/SwiftKafka/KafkaTopicConfig.swift

```diff
@@ -0,0 +1,121 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+
+/// When messages get published to a non-existent topic, a new topic is created using the `KafkaTopicConfig`
+/// configuration object (only works if server has `auto.create.topics.enable` property set).
```

Comment:
This seems like a very specific first sentence for this type. Maybe we can start with the more general docs here first and add this as a `- Note:` block

Reply:
Good point. I have moved it to the documentation of `KafkaProducer` as it makes more sense for the user (IMO).